android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -1998,6 +1998,17 @@ public static void removeKeyboardEventListener(OnKeyboardEventListener listener)
     KMKeyboard.removeOnKeyboardEventListener(listener);
   }
 
+  public static int getOrientation(Context context) {
+    Display display = ((WindowManager) context.getSystemService(Context.WINDOW_SERVICE)).getDefaultDisplay();
+    int rotation = display.getRotation();
+    if (rotation == Surface.ROTATION_0 || rotation == Surface.ROTATION_180) {
+      return Configuration.ORIENTATION_PORTRAIT;
+    } else if (rotation == Surface.ROTATION_90 || rotation == Surface.ROTATION_270) {
+      return Configuration.ORIENTATION_LANDSCAPE;
+    }
+    return Configuration.ORIENTATION_UNDEFINED;
+  }
```

Comment:
Did you find a reference for using this pattern?

Reply:
Here's one
https://developer.android.com/reference/android/view/Display#getRotation()

and a gist
https://gist.github.com/SammyVimes/92c0627195c4c55ea800